cmd/nomos/status/status.go

```diff
@@ -53,6 +54,7 @@ func init() {
 	Cmd.Flags().DurationVar(&pollingInterval, "poll", 0*time.Second, "Polling interval (leave unset to run once)")
 	Cmd.Flags().StringVar(&namespace, "namespace", "", "Namespace repo to get status for (multi-repo only, leave unset to get all repos)")
 	Cmd.Flags().BoolVar(&resourceStatus, "resources", true, "show resource level status for Namespace repo (multi-repo only)")
+	Cmd.Flags().StringVar(&rsyncname, "rsyncname", "", "rsyncname to filter rsync)")
```

Comment:
I don't think `rsyncname` is the right argument name here. We use this terminology informally to refer to RootSync|RepoSync, but I don't expect users to be familiar with this.

This also doesn't seem to give much control over filtering RootSync vs RepoSync. e.g. what if there are multiple RSyncs in the cluster that have the same `Name`? Perhaps as a user I would like to filter for RepoSyncs, or RootSyncs, or by a particular namespace, etc.

This also only allows for exact name matching. I imagine a user might want to support some form of pattern matching for more flexibility.

WDYT?

Reply:
yeah it might be better to use a different argument name, what would you suggest? Maybe something like `--resourceName`?

From what I understand, most feature request from the customer were about filtering for the RootSync that they are looking for. They typically find `--namespace` to be sufficient for RepoSync. In this PR I made the filter to work for both Root and Repo sync, but they can be separated, maybe with different flags for root and repo. What would you suggest to use as filter if not using the `Name`?

Reply:
I'd suggest an argument name such as `--name`

I think providing a `Name` filter makes sense, but an exact name matching filter on its own doesn't seem sufficient. At a minimum I think it should also support filtering by RootSync/RepoSync for this use case, e.g.
- an additional `sync-kind` type field that selects RootSync or RepoSync
- perhaps special handling for `--namespace=config-management-system` to list RootSyncs, although I'm not sure if this behavior might be confusing

Reply:
Agree with using `--name` as the argument name, but should it be a plural?

I think we should filter the RootSyncs and RepoSyncs with a combination  of `--names` and `--namespaces` together.

When only `--namespaces` is specified, it should list all RepoSyncs/RootSyncs in the namespaces.

When only `--names` is specified, it should list all RepoSyncs and RootSyncs with the name in all namespaces.

When both `--namespaces` and `--names` are specified, it should only list the matching ones.

Given that RepoSync can't be in the `config-managmeent-system` namespace, while RootSync can only be in the namespace, I feel like `--sync-kind` seems unnecessary.

Reply:
My only hesitation is that `--namespace` currently seems to implicitly mean "list RepoSyncs in the provided namespace". I agree it would make more sense if this allowed listing `RootSync` as well (assuming `config-management-system` is provided), this will just require a bit larger of a change to the behavior.

Other than that caveat, +1 to @nan-yu

Reply:
I take it back on the comments around `plural`. I thought the current `--namespace` is a plural, which supports listing RepoSyncs in multiple namespaces.

I think is is okay to use singular to filter with a particular namespace and/or name.

Reply:
One benefit of adding `--sync-kind` is that it can support listing all RepoSyncs in all namespaces, which can't be achieved with the current arguments.

But my concern is that if `--sync-kind` is not provided, do we infer it from `--namespace`, or do we force it to be specified when `--namespace` or `--name` is provided?

Reply:
Yeah the different combinations would add quite a bit of complexity. Might be easier to discuss in a doc

Reply:
i have created a onepager doc to continue the discussion, holding this PR for now

Reply:
the use case has been updated to follow the discussion here and in the design docs